### ISSUE TITLE
Planning 2.0

### ISSUE THREAD

**Issue by ming-codes** [lightblade/ember-cli-d3#47]
This issue is used to track/discuss any plan for 2.0 rollout. Below is a list of ideas that came to mind. Feedbacks appreciated.
- #35 `data-generator` is being removed. Users are suggested to migrate to using [Marak/faker.js](https://github.com/marak/Faker.js/)
- D3 4.0 release is imminent. 2.0 will migrate to using 4.0.
- `graph` test helper needs to be better thought out. With the introduction async/await, there may be better way to write this test helper.
- Drop support for Ember 1.x and only supporting Ember 2.x moving forward. With upcoming release of Ember LTS. It may make sense to only support LTS versions.
- ~~The [chiasm-project](https://github.com/chiasm-project) is interesting. Maybe good idea to base `{{data-visual}}` on it.~~
- Try to steer people to design data vis that take advantage of other great add-ons in the Ember community. This probably also means `ember-cli-d3` will try to do less while empowering developer to do more. From the news that I have gathered on what's coming down the pipeline for Ember and Ember CLI, there may not be a need for this addon in the future.

**Comment by michaelreynolds** [comment]
So does this mean it doesn't even run in Ember 2.x?

**Comment by ming-codes** [comment]
No, it does. https://travis-ci.org/ming-codes/ember-cli-d3/builds/117378226

This issue tracks planning of ember-cli-d3 2.0. Think of it as me thinking out loud.

**Comment by wolfbiter** [comment]
+1 to supporting D3 4.0

@ming-codes any updates on "From the news that I have gathered on what's coming down the pipeline for Ember and Ember CLI, there may not be a need for this addon in the future"? Would love to see your suggestions for how to migrate apps dependent on `ember-cli-d3` if you plan to discontinue this great repo

**Comment by ming-codes** [comment]
Hi @wolfbiter, thanks for the support. There are a few other add-ons that goes on different directions.
- [LocusEnergy/ember-d3-helpers](https://github.com/LocusEnergy/ember-d3-helpers)
- [ivanvanderbyl/maximum-plaid](https://github.com/ivanvanderbyl/maximum-plaid)

While both are great add-ons made by great people, they're both limited to building simple charts. This is something that libraries like HighCharts already solved and have done a great job solving it.

The initial goal of this add-on was to solve the `<base>` tag issue in ember-cli. This problem was since solved by ember-cli itself at version 2.8.

Since then I think the add-on went into a sort of feature creep, adding in layering to support canvas, WebGL, and HTML and selection proxy..which admittedly added more problem than it solves.

For 2.0, I think I'll be going off an entirely different direction to differentiate itself from other add-ons. At the same time, I need to scale down the scope to limit to only typical d3 usage.
